code/controllers/subsystems/overlays.dm

```diff
@@ -185,6 +185,24 @@ var/datum/controller/subsystem/overlays/SSoverlays
 	if(NOT_QUEUED_ALREADY)
 		QUEUE_FOR_COMPILE
 
+/atom/proc/set_overlays(list/overlays, priority = FALSE)	// Sets overlays to a list, equivalent to cut_overlays() + add_overlays().
+	if (!overlays)
+		return
+
+	overlays = build_appearance_list(overlays)
+
+	if (priority)
+		LAZYCLEARLIST(priority_overlays)
+		if (overlays)
```

Comment:
Can `build_appearance_list` return null? This is the only case where this check would be useful.